io/io/src/RRawFile.cxx

```diff
@@ -68,6 +90,23 @@ ROOT::Internal::RRawFile::Create(std::string_view url, ROptions options)
 #ifdef _WIN32
       return std::unique_ptr<RRawFile>(new RRawFileWin(url, options));
 #else
+      // If URL is a file on an EOS FUSE mount, attempt redirection to XRootD protocol.
+      if (gEnv->GetValue("TFile.CrossProtocolRedirects", 1) == 1) {
+         // At the moment the url parameter can only be the file name
+         ssize_t len = getxattr(url.data(), "eos.url.xroot", nullptr, 0);
+         if (len > 0) {
+            std::string xurl(len, 0);
+            std::string fileNameFromUrl{url.data()};
+            if (getxattr(fileNameFromUrl.c_str(), "eos.url.xroot", &xurl[0], len) == len) {
+               // Sometimes the `getxattr` call may return an invalid URL due
+               // to the POSIX attribute not being yet completely filled by EOS.
+               if (auto baseName = fileNameFromUrl.substr(fileNameFromUrl.find_last_of("/") + 1);
+                   std::equal(baseName.crbegin(), baseName.crend(), xurl.crbegin())) {
+                  return Create(xurl, options);
```

Comment:
I think we should check that the result actually has the XRootD protocol prefix. Otherwise a malfunctioning file system can send us in an endless loop.